apis/metal3.io/v1alpha1/baremetalhost_validation.go

```diff
@@ -53,3 +58,11 @@ func validateRAID(r *RAIDConfig) error {
 
 	return nil
 }
+
+func validateBMHName(bmhname string) error {
+	if strings.Contains(bmhname, "~") || bmhname == "" {
```

Comment:
BMH name emptiness check might belong to another PR, I'd prefer to use that way;

```
func validateBMHName(bmhname string) error {
if strings.Contains(bmhname, "~") {
		return fmt.Errorf("BareMetalHost resource name cannot be empty or contain the character `~`")
	}
return nil
}
```

Reply:
But wouldn't it make sense to keep all the invalid name validations together.

Reply:
According to the https://github.com/kubernetes/kubernetes/blob/4753fa5dee121308581ee5cc1a790a7b87315ed8/staging/src/k8s.io/apimachinery/pkg/apis/meta/v1/types.go#L124. BMH name can not be updated and it is optional.

When it is not given, it generates name https://github.com/kubernetes/kubernetes/blob/4753fa5dee121308581ee5cc1a790a7b87315ed8/staging/src/k8s.io/apimachinery/pkg/apis/meta/v1/types.go#L142

IMO, there is no need to check bmh name is nil or not.